pkg/crd/crd.go

```diff
@@ -0,0 +1,132 @@
+package crd
+
+import (
+	"context"
+	"encoding/json"
+
+	apixv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apixv1client "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/yaml"
+
+	"github.com/dominodatalab/distributed-compute-operator/config/crd"
+)
+
+var log = zap.New()
+
+// Apply will create or update all project CRDs inside a Kubernetes cluster.
+// The latest available version of the CRD will be used to perform this operation.
+func Apply(ctx context.Context) error {
+	apply := func(client apixv1client.CustomResourceDefinitionInterface, crd *apixv1.CustomResourceDefinition) error {
+		found, err := client.Get(ctx, crd.Name, metav1.GetOptions{})
+
+		switch {
+		case apierrors.IsNotFound(err):
+			log.Info("Creating CRD", "Name", crd.Name)
+			_, err = client.Create(ctx, crd, metav1.CreateOptions{})
+		case err == nil:
```

Comment:
You use a different pattern elsewhere:
```
if apiErrors.IsNotFound(err) { ... }
if err != nil { ... }

...
```

Not sure which is best, but a single style would be better I think.

Reply:
I totally agree. I'll converge on a single pattern.

Reply:
done